connectors/sources/sharepoint_online.py

```diff
@@ -1627,30 +1628,36 @@ async def _download_content(self, download_func, original_filename):
         source_file_name = ""
         file_extension = os.path.splitext(original_filename)[-1]
 
-        async with NamedTemporaryFile(
-            mode="wb", delete=False, suffix=file_extension
-        ) as async_buffer:
-            # download_func should always be a partial with async_buffer as last argument that is not filled by the caller!
-            # E.g. if download_func is download_drive_item(drive_id, item_id, async_buffer) then it
-            # should be passed as partial(download_drive_item, drive_id, item_id)
-            # This way async_buffer will be passed from here!!!
-            await download_func(async_buffer)
-
-            source_file_name = async_buffer.name
-
-        if self.configuration["use_text_extraction_service"]:
-            body = ""
-            if self.extraction_service._check_configured():
-                body = await self.extraction_service.extract_text(
-                    source_file_name, original_filename
+        try:
+            async with NamedTemporaryFile(
+                mode="wb", delete=False, suffix=file_extension
+            ) as async_buffer:
+                # download_func should always be a partial with async_buffer as last argument that is not filled by the caller!
+                # E.g. if download_func is download_drive_item(drive_id, item_id, async_buffer) then it
+                # should be passed as partial(download_drive_item, drive_id, item_id)
+                # This way async_buffer will be passed from here!!!
+                await download_func(async_buffer)
+
+                source_file_name = async_buffer.name
```

Comment:
I think it should be above `download_func` as `download_func` can raise an error and this line won't be reached